apps/bfd-pipeline/bfd-pipeline-rif-load/src/main/java/gov/cms/bfd/pipeline/rif/load/RifLoader.java

```diff
@@ -512,8 +552,7 @@ private void processAsync(
        * ensure that the failing data is captured.
        */
       try {
-        if (entityManager != null && entityManager.getTransaction().isActive())
-          entityManager.getTransaction().rollback();
+        if (txn != null && txn.isActive()) txn.rollback();
```

Comment:
it looks like this `txn` isn't in the scope of this `finally`, since it's set during the `try`. I could just be counting the brackets wrong.

Reply:
The idea behind this change is not to rely on the fact that getTransaction() will return the same transaction as the one at the start of the transaction. Instead of doing...

```
try {
  em.getTransaction().begin();
  ...
  em.getTransaction().commit();
} finally {
  em.getTransaction().rollback();
}
```

We write...
```
EntityTransaction txn = null;
try {
  txn = em.getTransaction();
  txn.begin();
  ...
  txn.commit();
} finally {
  if (txn != null) txn.rollback();
}
```
Not as elegant, but I think it is safer. What do you think? Granted, the function is a mess, as the TODO states. What do you think?

Reply:
Definitely like the change to clearly use the same transaction all the way through. My question is really more about java scope in general: since `txn` is set to `em.getTransaction()` inside the `try` block does it retain that value so that `txn.rollback()` can be called in the `finally`?

Reply:
answered my own question looks like it [does retain the value](https://repl.it/repls/EnchantedOpulentIrcbot) out of the `try` block.

Reply:
Declarations are block-scoped, not assignments/values.